### PR TITLE
feat: add eslint ci, fix eslint errors

### LINKED ISSUE #23: Integrate eslint checking CI

### PR DESCRIPTION
**Pull request by roberthovsepyan**
Closes #23

### CODE REVIEW

File: src/parsing/index.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ export interface KeywordSuggestion {
     weight: number;
 }
 
-export interface SyntaxError {
+export interface ParserSyntaxError {
```
Comment (NikitaShkaruba) [inline src/parsing/index.ts:117]: Why do we need to rename it?

Comment (roberthovsepyan) [reply src/parsing/index.ts:117]: because SyntaxError is a built-in type

File: package.json [hunk 2]
```diff
@@ -58,7 +58,7 @@
       "prettier --write"
     ],
     "*.{js,ts}": [
-      "eslint --fix"
+      "eslint src --fix"
```
[the remark this answers is not in the document]
Comment (roberthovsepyan) [reply package.json:61]: Nope, it only lints js, ts files

**Review by NikitaShkaruba — APPROVED** [review]
We needed that badly!

**Pull request merged**